projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js

```diff
@@ -6,220 +6,285 @@ import { Placeholder, Button, Spinner } from '@wordpress/components';
-			title,
-			content
-		);
-	}
+	const categoryObjects = useSelect(
+		select => {
+			return categories
+				.map( categoryId => {
+					const category = select( 'core' ).getEntityRecord( 'taxonomy', 'category', categoryId );
+
+					if ( ! category ) {
+						// Data is not yet loaded
+						loading = true;
+						return;
+					}
 
-	return sprintf(
-		/** translators: This will be a prompt to OpenAI to generate a post based on the post title */
-		__( 'Write content of a post titled "%s"', 'jetpack' ),
-		title
+					return category;
+				} )
+				.filter( Boolean ); // Remove undefined values
+		},
+		[ categories ]
 	);
-}
 
-/**
- * Gather data available in Gutenberg and prepare the best prompt we can come up with.
- *
- * @param {Function} select - function returning Gutenberg data store.
- * @returns {string} - prompt ready to pipe to OpenAI.
- */
-function preparePromptBasedOnEditorState( select ) {
-	const editorContent = formatEditorCanvasContentForCompletion(
-		select( 'core/block-editor' )
-	).slice( -240 );
+	const tags = useSelect( select => select( 'core/editor' ).getEditedPostAttribute( 'tags' ), [] );
 
-	// Let's grab post data so that we can do something smart.
-	const currentPost = select( 'core/editor' ).getCurrentPost();
-	if ( ! currentPost ) {
-		return '';
-	}
+	const tagObjects = useSelect(
+		select => {
+			return tags
+				.map( tagId => {
+					const tag = select( 'core' ).getEntityRecord( 'taxonomy', 'post_tag', tagId );
 
-	// If there is no title, there is not much we can do.
-	if ( ! currentPost.title ) {
-		return createPrompt( '', editorContent, '' );
-	}
+					if ( ! tag ) {
+						// Data is not yet loaded
+						loading = true;
+						return;
+					}
 
-	// We are filtering out the default "Uncategorized"
-	const categories = currentPost.categories.filter( catId => catId !== 1 );
-	const tags = currentPost.tags;
+					return tag;
+				} )
+				.filter( Boolean ); // Remove undefined values
+		},
+		[ tags ]
+	);
 
-	// User did not set any categories, we are going to base the suggestions off a title.
-	if ( ! categories.length && ! tags.length ) {
-		return createPrompt( currentPost.title, editorContent, '' );
-	}
+	useEffect( () => {
+		setIsLoadingCategories( loading );
+	}, [ loading ] );
 
-	// We are grabbing more data from WP.
-	const categoryObjects = categories.map( categoryId =>
-		select( 'core' ).getEntityRecord( 'taxonomy', 'category', categoryId )
-	);
-	const tagObjects = tags.map( tagId =>
-		select( 'core' ).getEntityRecord( 'taxonomy', 'post_tag', tagId )
-	);
+	const createPrompt = () => {
+		const content = contentBefore
+			.filter( function ( block ) {
+				return block && block.attributes && block.attributes.content;
+			} )
+			.map( function ( block ) {
+				return block.attributes.content.replaceAll( '<br/>', '\n' );
+			} )
+			.join( '\n' );
+		const shorter_content = content.slice( -240 );
```

Comment:
Should we add a comment here why `-240`. Should we store this in a const?

Reply:
Done